passes/memory/memory_bram.cc

```diff
@@ -1100,6 +1136,42 @@ void handle_cell(Cell *cell, const rules_t &rules)
 				goto next_match_rule;
 			}
 
+			for (auto iter: match.attr_unmatch) {
+				auto it = cell->attributes.find(iter.first);
+
+				if (it != cell->attributes.end()) {
+						log("    Rule for bram type %s is rejected: requirement 'attribute %s' is met.\n",
+								log_id(match.name), iter.first.c_str());
+						goto next_match_rule;
+				}
+				continue;
+			}
+
+			if (!match.attr_match.empty()) {
+				for (auto iter: match.attr_match) {
+					int len=std::tuple_size<decltype(iter)>::value;
+					auto it = cell->attributes.find(iter.first);
+
+					if (iter.second.empty()) {
+						log("    Rule for bram type %s is rejected: requirement 'attribute %s=\"%s\"' not met.\n",
+								log_id(match.name), iter.first.c_str(), iter.second.decode_string().c_str());
+						continue;
+					}
+
+					if (it != cell->attributes.end()) {
+						if (it->second == iter.second)
```

Comment:
The PR #1501  copies the value of the attribute, if given, so this comparison can be made easily. In the case of,`synthesis, ram_block`, no value is assigned to the attribute, therefore, `memory_collect.cc` adds a value of 1, meaning the attribute is present. In that case, the comparison is wrong, since data types are different.